reactor-core/src/test/java/reactor/core/publisher/FluxSwitchOnFirstTest.java

```diff
@@ -580,23 +766,23 @@ public void shouldBeAbleToCancelSubscription() throws InterruptedException {
                 publisher.next(1L);
 
                 switchTransformed.subscribe(captureElement::set,
-                        __ -> {
-                        },
+                        __ -> { },
                         () -> captureCompletion.set(true),
                         s -> new Thread(() -> RaceTestUtils.race(publisher::complete,
                                 () -> RaceTestUtils.race(s::cancel,
                                         () -> s.request(1),
                                         Schedulers.parallel()),
                                 Schedulers.parallel())).start());
 
-                Assert.assertTrue(latch.await(5, TimeUnit.SECONDS));
-                Assert.assertThat(requested.get(), equalTo(1L));
+                Assertions.assertThat(latch.await(5, TimeUnit.SECONDS)).isTrue();
+                Assertions.assertThat(requested.get()).isEqualTo(1L);
                 capturedElements.add(captureElement.get());
                 capturedCompletions.add(captureCompletion.get());
             }
 
-            Assume.assumeThat(capturedElements, hasItem(equalTo(0L)));
-            Assume.assumeThat(capturedCompletions, hasItem(equalTo(false)));
+
+            Assumptions.assumeThat(capturedElements).contains(0L);
```

Comment:
No, I don't. In this case, it could happen that racing can end up with all completion instead of cancellation. So I don't want to fail the test because for some system reasons element propagation was faster than the Thread allocation and cancellation task execution

Reply:
it is kind of unstable test, so I decided to use assume there

Reply:
please document that very explicitly with a comment then!

Reply:
sure buddy

Reply:
fixed